sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/ItemBulkOperation.java

```diff
@@ -105,6 +110,15 @@ JsonSerializable getSerializedOperationInternal() {
         return jsonSerializable;
     }
 
+    public void setIndex(int index) {
+        if (this.index == 0) {
```

Comment:
why the check for 0? to only write it once when it is initially 0?

Reply:
Yes to only set it once. If not, it is possible that the index will be changed when the operation fails due to a partition split.

Reply:
Can we explicitly set `this.index` to 0 in the constructor - for readability reasons?

Reply:
Added it to constructor and left a comment to explain why we do that check.